index.js

```diff
@@ -25,13 +25,31 @@ var UNICODE_CHARS = {
     '\u2029': '\\u2029'
 };
 
+// We can‘t just instanceof Date since dates are already converted to strings
+// because of native Date.prototype.JSON (which use toISOString)
+var DATE_LENGTH = new Date().toISOString().length
+function isDate(d) {
+  try {
+    return (
+      // testing length first to avoid new Date() for every string
+      d.length === DATE_LENGTH &&
```

Comment:
> I forgot that we can check back into the original value if we want to.

Are you sure about that? How does it works for key in depth?! (eg: `{ some: { thing: { d: `, key will be `d`, so `currentObj[key]` will be inaccurate).